boot-qemu.sh

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env bash
+
+# Root of the repo
+BASE=$(cd "$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")" && pwd)
+
+
+# Prints an error message in bold red then exits
+function die() {
+    printf "\n\033[01;31m%s\033[0m\n" "${1}"
+    exit 1
+}
+
+
+# Check that a binary is found
+function checkbin() {
+    command -v "${1}" &>/dev/null || die "${1} could not be found, please install it!"
+}
+
+
+# Parse inputs to the script
+function parse_parameters() {
+    while (( ${#} )); do
+        case ${1} in
+            -a|--arch|--architecture)
+                shift
+                case ${1} in
+                    arm32_v5|arm32_v6|arm32_v7|arm64|mips|mipsel|ppc32|ppc64|ppc64le|x86_64) ARCH=${1} ;;
+                    *) die "Invalid --arch value '${1}'" ;;
+                esac ;;
+
+            -d|--debug)
+                set -x ;;
+
+            -h|--help)
+                echo
+                cat "${BASE}"/boot-qemu-help.txt
+                echo
+                exit 0 ;;
+
+            -i|--interactive|--shell)
+                INTERACTIVE=true ;;
+
+            -k|--kbuild-folder)
+                shift && KBUILD_DIR=${1} ;;
+
+            -t|--timeout)
+                shift && TIMEOUT=${1} ;;
+
+            *)
+                die "Invalid parameter '${1}'" ;;
+        esac
+        shift
+    done
+}
+
+
+# Sanity check parameters and required tools
+function sanity_check() {
+    # Kernel build folder and architecture are required paramters
+    # We are not interactive by default and the default timeout is 3m
+    : "${ARCH:?}" "${KBUILD_DIR:?}" "${INTERACTIVE:=false}" "${TIMEOUT:=3m}"
+
+    # KBUILD_DIR could be a relative path; turn it into an absolute one with readlink
+    KBUILD_DIR=$(readlink -f "${KBUILD_DIR}")
+
+    # Let the user know if the kernel build folder does not exist
+    [[ -d ${KBUILD_DIR} ]] || die "${KBUILD_DIR} does not exist!"
+
+    # Make sure zstd is install
+    checkbin zstd
+}
+
+
+# Decompress rootfs images
+function decomp_rootfs() {
+    [[ ${ARCH} =~ arm32 ]] && ARCH_RTFS_DIR=arm
```

Comment:
please add a comment that this is because of our convention for arm32, yet we can share images between them.

Reply:
Will do.